__init__.py

```diff
@@ -15,7 +15,38 @@ class MarkExperience(Event):
 
 
 MODIFIER_REGEX = "[+-]?[0,1,2,3]"
-STAT_REGEX = f"(?:(?:!cool|!tough|!sharp|!charm|!weird) {MODIFIER_REGEX})"
+GAME_STATS = {"motw": ["cool", "tough", "sharp", "charm", "weird"],
+              "pbtastartrek": ["aggressive", "bold", "talk", "tech"]}
+STAT_REGEXES = {}
+
+@match_regex(r"\!set game ?(?P<gamename>.*)", case_sensitive=False)
+@memory_in_event_room
+async def set_game(opsdroid, config, message):
+    game = message.regex.capturesdict()['gamename'][0]
+    if game not in GAME_STATS.keys():
+        message.respond(f"I don't know how to play that. Available options are: {', '.join(GAME_STATS.keys())}")
+        return
+    await opsdroid.memory.put(f"pbta_stats", GAME_STATS[game])
+
+
+async def get_stats(room):
+    return await opsdroid.memory.get("pbta_stats", [])
+
+
+def html_list(sequence):
+    html_stats = ''.join([f"<li>{s.capitalize()}</li>" for s in sequence])
+    return f"<ul>{html_stats}</ul>"
+
+
+async def filter_by_game_stats(string, room):
+    if room not in STAT_REGEXES.keys():
+        gamestats = await get_stats(room)
+        if not gamestats:
+            return []
+        stats_re = regex.compile(f"(?:(?:{'|'.join(['!'+s for s in gamestats])}) {MODIFIER_REGEX})")
+        STAT_REGEXES[room] = stats_re
+    stats = regex.findall(STAT_REGEXES[room], string, flags=regex.IGNORECASE)
```

Comment:
I think you can now do this:

```suggestion
    stats = STAT_REGEXES[room].findall(string, flags=regex.IGNORECASE)
```